source/core.py

```diff
@@ -1055,7 +1056,8 @@ def _doPostNvdaStartupAction():
 			" This likely indicates NVDA is exiting due to WM_QUIT.",
 		)
 		queueHandler.pumpAll()
-	_terminate(remoteClient)
+	if remoteClient.remoteRunning():
+		_terminate(remoteClient)
```

Comment:
Since `terminate` is already a smart function that doesn't terminate when not running, you can leave out the if check here.